src/sagemaker/model.py

```diff
@@ -635,7 +648,7 @@ def prepare_container_def(
             self.repacked_model_data or self.model_data,
             deploy_env,
             image_config=self.image_config,
-            accept_eula=getattr(self, "accept_eula", None),
+            accept_eula=accept_eula or getattr(self, "accept_eula", None),
```

Comment:
question: Why not adhere the changes to `create()` to the current pattern outlined in [deploy()](https://github.com/aws/sagemaker-python-sdk/blob/ef8dd31d6b5e915c746788e9ec3cdb308f227dd2/src/sagemaker/model.py#L1366)? I'd recommend we stick to either passing in the value or through using `getattr`, but not both. wdyt?

Reply:
imo we should pass it in and modify `deploy()` to match but would definitely want to have the group weigh in on this

Reply:
question: what happens if `accept_eula` is `False` ? Does it use the attribute?

Could you add a unit test for this case please?

Reply:
Yeah, totally agree @evakravi. Doesn't have to be this PR but can we put a backlog item to refactor `deploy()` to match and remove `getattr` & `self.accept_eula`?